daemon/cluster/noderunner.go

```diff
@@ -271,7 +271,13 @@ func (n *nodeRunner) enableReconnectWatcher() {
 		if n.stopping {
 			return
 		}
-		config.RemoteAddr = n.cluster.getRemoteAddress()
+		remotes := n.cluster.getRemoteAddressList()
+		if len(remotes) > 0 {
+			config.RemoteAddr = remotes[0]
+		} else {
+			config.RemoteAddr = ""
+		}
```

Comment:
@tonistiigi: There are two cases here. Either the node had successfully joined the cluster, or not.

If it has joined the cluster, `JoinAddr` is unnecessary and will be ignored.

If it hasn't joined the cluster, `getRemoteAddress` will return the original `JoinAddr`, because `Remotes` was only populated with that address.

So I don't see why overwriting `RemoteAddr` is helpful.